runtime/src/mm/cpp/Memory.cpp

```
@@ -74,6 +74,20 @@ extern "C" void RestoreMemory(MemoryState*) {
     // TODO: Remove when legacy MM is gone.
 }
 
+extern "C" RUNTIME_NOTHROW OBJ_GETTER(AllocInstance, const TypeInfo* typeInfo) {
+    auto* threadData = mm::ThreadRegistry::Instance().CurrentThreadData();
+    auto* object = threadData->objectFactoryThreadQueue().CreateObject(typeInfo);
+    RETURN_OBJ(object);
+}
+
+extern "C" OBJ_GETTER(AllocArrayInstance, const TypeInfo* typeInfo, int32_t elements) {
+    RuntimeAssert(elements >= 0, "elements cannot be negative");
```

It should throw an exception if `elements < 0`.